### DISCUSSION TITLE
Deploying Express Server on Vercel but only getting Index.JS to show

### DISCUSSION THREAD

**Discussion by Aborgono** [vercel/community#4670] [Help]
### Question

Hello I've gone through previous discussion on this same topic but given previous answer I am not able to solve my issue. I have deployed a Express Hosted backend server on Vercel and my page shows my index.js configs. Steps I've taken: Ive created an api folder, created index.js, create vercel.json with rewrite script, I have changed paths to match configuration correctly, and I have added a gitignore for node_modules. Here is my index.js: 
const express = require ('express')
const app = express()
const cors = require ('cors')
const PORT = 8080
// const mongoose = require ('mongoose')

const userRoute = require('../routes/user')
const leaderBoardRoute = require('../routes/leaderBoard')
const bodyParser = require('body-parser')

app.use(cors())
app.use(bodyParser.json())
app.use('/users', userRoute)
app.use('/leaderBoard', leaderBoardRoute)

app.listen(PORT, () => {
    console.log(`App is running on ${PORT}`);
})
// Please let me know if you have other questions

**Comment by amyegan** [comment]
Hey @Aborgono! We have a guide available to help you [use Express.js with Vercel](https://vercel.com/guides/using-express-with-vercel). The set up is a little different from a more traditional server environment. The guide also covers some of the [drawbacks and edge cases](https://vercel.com/guides/using-express-with-vercel#drawbacks-and-edge-cases) to consider.

https://vercel.com/guides/using-express-with-vercel

**Reply by Aborgono** [reply]
I have read that article as well as other discussion regarding my issue. I am using a standalone express backend. Despite following these steps, I cannot get it to work. The only difference I see that may be causing my issue is that I am using mongoDB as the database...? I've made sure to update all dependencies as well as re-install node modules and my backend will still not work.

**Reply by amyegan** [reply]
Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? I have an example Express.js app that I built based on that guide and I wasn't able to reproduce your problem. 

<img width="236" alt="Express app file structure showing index file inside API folder" src="https://github.com/vercel/community/assets/8247105/6bf7a737-3a44-41e8-8e08-c3400fcd643a">

**Reply by Aborgono** [reply]
Yes, I do. I had it working when I had my index.js in the src directory. I just moved it to the api folder and now I am getting a 404 error. Note: this is in the minimal reproducible example of mine. Note: I am using js not ts

**Reply by Aborgono** [reply]
UPDATE: My api is now working. I changed my vercel.json to a structure you had in one of your respositories (express-1663). However I still cannot access my routes. Two questions. is my folder structure correct? and do i need to create more configuration in my vercel.json for the routes? 
![image](https://github.com/vercel/community/assets/127222704/dbbdcd2d-f2af-45e9-b4ef-a7c40701d9e8)

**Comment by Aborgono** [comment]
I figured out my issue if anyone has the same problem. I went to integrations and installed mongoDB, this made it so I could connect to my atlas database.
